### PR TITLE
Schema API reference

### LINKED ISSUE #576: Change datasource db convention
```
datasource postgresql {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```

to

```
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```

### LINKED ISSUE #605: Schema API reference

### LINKED ISSUE #626: Datasource provider array

### LINKED ISSUE #688: Schema can use defined order of fields (vs. alphabetically)
2.3.0 will change Introspection to sort the columns/fields by the order in the database instead of alphabetically. That means that all the schema examples in the documentation now also theoretically can default to having a more defined order of fields (e.g. `id` first, then grouped by meaning or similar), as that will be what our users will experience.

### PR DESCRIPTION
**Pull request by mhwelander**
* [ ] Redirect `models` to `data-model`

### CODE REVIEW

**Comment by mhwelander** [comment]
Closes #576

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/03-generators.mdx [hunk 1]
```diff
@@ -4,158 +4,50 @@ metaTitle: 'Generators (Reference)'
...
-| Build OS              | Build name             | OpenSSL |
-| :-------------------- | :--------------------- | :-----: |
-| Arch Linux 2019.09.01 | `debian-openssl-1.1.x` |  1.1.x  |
-
-#### The native binary target
-
-The `native` binary target is special. It doesn't map to a concrete operating system. Instead, when `native` is specified in `binaryTargets`, Prisma detects the _current_ operating system and automatically specifies the correct binary target for it.
-
-As an example, assume you're running **Mac OS** and you specify the following generator:
+Each Prisma schema has one generator, represented by the [`generator`](prisma-schema-reference#generator) <span class="api"></span> block:
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/03-generators.mdx]:
```suggestion
A Prisma schema can have one or more generators, represented by the [`generator`](prisma-schema-reference#generator) <span class="api"></span> block:
```

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/03-generators.mdx [hunk 2]
```diff
@@ -4,158 +4,50 @@ metaTitle: 'Generators (Reference)'
...
 ---
 
-## Overview
+<TopBlock>
 
-A generator can be specified via the `generator` block in the Prisma schema. Generators in your Prisma schema specify what assets are generated when the `prisma generate` command is invoked.
+A generator determines which assets are created when you run the `prisma generate` command, including:
 
-## Fields
+* The **client language** (currently only `prisma-client-js` is available)
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/03-generators.mdx]:
```suggestion
* The **Prisma client (language specific)** (currently only `prisma-client-js` is available)
```

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/03-generators.mdx [hunk 3]
```diff
@@ -4,158 +4,50 @@ metaTitle: 'Generators (Reference)'
...
+* Target **engine binaries** (for example, `debian-openssl-1.0.x` if you are deploying to Ubuntu 14.04, or `native` if you are working locally)
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/03-generators.mdx]: This is technically only a configuration of the `prisma-client-js` and does not apply to other generators.

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/03-generators.mdx [hunk 4]
```diff
@@ -4,158 +4,50 @@ metaTitle: 'Generators (Reference)'
...
 
-This example shows how to use a custom generator that's located in a directory called `my-generator`.
+### Custom binary targets
 
-```prisma
-generator client {
-  provider      = "./my-generator"
-}
-```
+You can use environment variables to [specify a custom binary](../../../understand-prisma/under-the-hood#using-custom-engine-binaries) for one or more engines.
```
Comment (janpio) [inline content/03-reference/01-tools-and-interfaces/01-prisma-schema/03-generators.mdx]: Technically the complete "Binary targets" section is documentation of Prisma Client JS, which let's you influence those via a configuration property in the `generator` block.

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/03-generators.mdx (hunk 3 above)
Comment (mhwelander) [reply content/03-reference/01-tools-and-interfaces/01-prisma-schema/03-generators.mdx]: Updated to say specifically `prisma-client-js`

File: content/03-reference/01-tools-and-interfaces/01-prisma-schema/03-generators.mdx (hunk 4 above)
Comment (mhwelander) [reply content/03-reference/01-tools-and-interfaces/01-prisma-schema/03-generators.mdx]: Updated to specify `prisma-client-js` next to `binaries`

**Pull request merged**